### DISCUSSION TITLE
Starting Action as service account user?

### DISCUSSION THREAD

**Discussion by slad-cloudapp** [community/community#35361] [Actions]
Hello,

Is there a way to start GitHub actions pipeline as specific user? I have pipeline which is triggered on merge to main branch but it looks like when the Actions pipeline would run it will trigger it as original author who merged the PR. I would like the pipeline to run with service account user in such scenarios, is that possible?

Reason: Pipeline bumps the version in package.json and pushes that change back to remote. I have branch protection rules that I want to enforce with the exception of this service account user. 

Any suggestions/ideas?

**Comment by slad-cloudapp** [comment]
Re-commenting to see if there are any thoughts/suggestions on this? Or this not supported at all?

**Comment by acnicholls** [comment]
Do you mean on a self-hosted server?

**Reply by acnicholls** [reply]
I checked the code for the runner, and there are only 3 allowed service accounts that you can use without a password

https://github.com/actions/runner/blob/a1244d22692e44d7e1fe36e6eee7773f065e4ed3/src/Runner.Listener/Configuration/NativeWindowsServiceHelper.cs#L368

NT Authority\NETWORK SERVICE
NT Authority\Local System
NT Authority\Local Service

any other service account will just jump to the password request.  

This might be able to be modified, because I found that my service account wasn't accepted as a valid username on the system until I entered it with a "$" symbol at the end of the name, like DOMAIN\ServiceAccountName$, instead of DOMAIN\ServiceAccountName.  So the $ could be checked for and the password prompt skipped (since it won't be needed anyway).  I'm going to submit a pull request.
